src/librustc/hir/lowering.rs

```diff
@@ -2968,11 +2972,18 @@ impl<'a> LoweringContext<'a> {
         bounds.iter().map(|bound| self.lower_param_bound(bound, itctx.reborrow())).collect()
     }
 
-    fn lower_block(&mut self, b: &Block, targeted_by_break: bool) -> P<hir::Block> {
+    fn lower_block_and_prepend_statements(
+        &mut self,
+        b: &Block,
+        targeted_by_break: bool,
+        f: impl FnOnce(&mut Vec<hir::Stmt>),
```

Comment:
I think it would be cleaner to just make this a `Vec<hir::Stmt>` instead of a callback. Then you can construct it normally, without `.insert(0)`.

Reply:
Fixed.